java/system-test/src/main/java/sleeper/systemtest/ingest/batcher/InvokeIngestBatcher.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.systemtest.ingest.batcher;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.sqs.AmazonSQS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.lambda.LambdaClient;
+
+import sleeper.clients.util.PollWithRetries;
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.configuration.properties.table.TablePropertiesProvider;
+import sleeper.configuration.properties.validation.BatchIngestMode;
+import sleeper.ingest.batcher.FileIngestRequest;
+import sleeper.ingest.batcher.IngestBatcherStore;
+import sleeper.ingest.batcher.store.DynamoDBIngestBatcherStore;
+import sleeper.ingest.job.status.IngestJobStatus;
+import sleeper.ingest.job.status.IngestJobStatusStore;
+import sleeper.ingest.status.store.job.DynamoDBIngestJobStatusStore;
+import sleeper.job.common.QueueMessageCount;
+import sleeper.systemtest.util.InvokeSystemTestLambda;
+import sleeper.systemtest.util.WaitForQueueEstimate;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import static java.util.function.Predicate.not;
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.INGEST_JOB_QUEUE_URL;
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.INGEST_LAMBDA_FUNCTION;
+import static sleeper.configuration.properties.table.TableProperty.INGEST_BATCHER_INGEST_MODE;
+import static sleeper.configuration.properties.table.TableProperty.INGEST_BATCHER_MAX_JOB_FILES;
+
+public class InvokeIngestBatcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InvokeIngestBatcher.class);
+
+    private final InstanceProperties instanceProperties;
+    private final Consumer<Consumer<TableProperties>> updateTableProperties;
```

Comment:
We could define an interface for this consumer to clarify how it's supposed to work.

Reply:
Part of the reason I added this was to make it testable but we didn't write any unit tests.